src/main/java/org/opensrp/repository/postgres/PlanRepositoryImpl.java

```diff
@@ -0,0 +1,209 @@
+
+/**
+ * Created by Vincent Karuri on 02/05/2019
+ */
+
+@Repository
+public class PlanRepositoryImpl extends BaseRepositoryImpl<PlanDefinition> implements PlanRepository {
+
+    @Autowired
+    private CustomPlanMapper planMapper;
+
+    @Autowired
+    private PlanMetadataMapper planMetadataMapper;
+    
+    @Override
+    public PlanDefinition get(String id) {
+        if (StringUtils.isBlank(id)) {
+            return null;
+        }
+
+        PlanExample planExample = new PlanExample();
+        planExample.createCriteria().andIdEqualTo(id).andDateDeletedIsNull();
+
+        List<Plan> pgPlan = planMapper.selectByExample(planExample);
+        List<PlanDefinition> plan = convert(pgPlan);
+
+        return isEmptyList(plan) ? null : plan.get(0);
+    }
+
+    @Override
+    public void add(PlanDefinition plan) {
+        if (getUniqueField(plan) == null) {
+            return;
+        }
+
+        if (retrievePrimaryKey(plan) != null) { 
+            return; // plan already added
+        }
+
+        Plan pgPlan = convert(plan);
+        if (pgPlan == null) {
+            return;
+        }
+
+        int rowsAffected = planMapper.insert(pgPlan);
+        if (rowsAffected < 1) {
+            return;
+        }
+        insertPlanMetadata(plan);
+    }
+
+    @Override
+    public void update(PlanDefinition plan) {
+        // todo: should we update deleted plans?
+        if (getUniqueField(plan) == null) {
+            return;
+        }
+
+        String id = retrievePrimaryKey(plan);
+        if (id == null) {
+            return; // plan does not exist
+        }
+
+        Plan pgPlan = convert(plan);
+        if (pgPlan == null) {
+            return;
+        }
+
+        int rowsAffected = planMapper.updateByPrimaryKey(pgPlan);
+        if (rowsAffected < 1) {
+            return;
+        }
+
+        updatePlanMetadata(plan);
+    }
+
+    @Override
+    public List<PlanDefinition> getAll() {
+        PlanExample planExample = new PlanExample();
+        planExample.createCriteria().andDateDeletedIsNull();
+        List<Plan> plans = planMapper.selectMany(planExample, null,0, DEFAULT_FETCH_SIZE);
+        return convert(plans);
+    }
+
+    @Override
+    public void safeRemove(PlanDefinition plan) {
+        if (plan == null) {
+            return;
+        }
+
+        String id = retrievePrimaryKey(plan);
+        if (id == null) {
+            return;
+        }
+
+        Plan pgPlan = convert(plan);
+        pgPlan.setDateDeleted(new Date());
+        planMapper.updateByPrimaryKey(pgPlan);
+    }
+
+    public List<PlanDefinition> getPlansByServerVersionAndOperationalArea(Long serverVersion, String operationalAreaId) {
+        PlanExample planExample = new PlanExample();
+        planExample.createCriteria().andServerVersionGreaterThanOrEqualTo(serverVersion).andDateDeletedIsNull();
+        List<Plan> plans = planMapper.selectMany(planExample, operationalAreaId, 0, DEFAULT_FETCH_SIZE);
```

Comment:
Rename `operationalAreaId` to `operationalAreaIds` where it can a single value or a comma delimited string of ids

Reply:
Done.